### ISSUE TITLE
Port twisted.internet.test.test_tcp and connectionmixins to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6069]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6069|
|Type|enhancement|
|Created|2012-10-05 19:52:17Z|
|Branch|https://github.com/twisted/twisted/tree/tcp-py3-6069-3|

In the interest of making [#6002](https://github.com/twisted/twisted/issues/6002) smaller, `twisted.internet.test.test_tcp` and `twisted.internet.test.connectionmixins` should be ported to Python 3 on their own.

<details><summary>Searchable metadata</summary>

```
trac-id__6069 6069
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_tcp_py3_6069_3 branches/tcp-py3-6069-3
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1349466737000000 1349466737000000
changetime__1350400924000000 1350400924000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [35994]) Branching to 'tcp-py3-6069'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst commented|
|-|-|

(In [36084]) Branching to 'tcp-py3-6069-2'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst commented|
|-|-|

(In [36157]) Branching to 'tcp-py3-6069-3'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst set owner to @exarkun|
|-|-|

Ready for review, I think. http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/tcp-py3-6069-3

Only non-test code change was changing order of checks in `abstract.py` so that you don't compare `ConnectionLost` to an integer, which fails on Python 3.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @itamarst|
|-|-|

Thanks.

  1. Maybe `connect` in twisted/internet/test/test_tcp.py should have a docstring, at least describing what shape `destination` should have, now that it is less self-describing in the signature.
  1. Add a reference to [#6070](https://github.com/twisted/twisted/issues/6070) to the `chr` function in `posixifaces.py`.

That's all.  Please merge after fixing the above.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_7"></a>|@itamarst set status to `closed`|
|-|-|

(In [36190]) Merge tcp-py3-6069-3.

Author: itamar
Review: exarkun
Fixes: [#6069](https://github.com/twisted/twisted/issues/6069)

twisted.internet.test.test_tcp is now ported to Python 3.
